src/viam/rpc/dial.py

```diff
@@ -280,7 +286,7 @@ async def _dial_direct(address: str, options: Optional[DialOptions] = None) -> C
 
         # Test if downgrade is required.
         downgrade = False
-        with socket.create_connection((host, port)) as sock:
+        with socket.create_connection((host, port), timeout=opts.timeout) as sock:
```

Comment:
This is only a test. The real socket connection happens in rust-utils. You'll probably have to change some rust-utils code to accept timeout as a parameter